logstash-core/lib/logstash/api/modules/node_stats.rb

```diff
@@ -13,47 +13,19 @@ class NodeStats < ::LogStash::Api::Modules::Base
           @stats = factory.build(:stats)
         end
 
-        # Global _stats resource where all information is
-        # retrieved and show
-        get "/" do          
+        get "/?:filter?" do
+          selected_fields = extract_fields(params["filter"].to_s.strip)
           payload = {
             :events => events_payload,
             :jvm => jvm_payload,
             :process => process_payload,
-            :mem => mem_payload
+            :mem => mem_payload,
+            :pipeline => pipeline_payload
           }
-
+          payload.select! { |k,v| selected_fields.include?(k) } unless selected_fields.empty?
```

Comment:
We use this code in two places. Maybe we could make this into a custom helper or add it into `respond_with` by making it take an additional argument?